### PR TITLE
Adds IsInList Endpoint and Supports Pagination on List Endpoint

### PR DESCRIPTION
**Pull request by TheGuyWhoCodes**
## MediaListItemServlet
 - Adds field "offset" for paginating through queue results, supports up to 24 at a time.
 - Throws 400 if missing

## MediaItemInServlet
 - Servlet that takes a userId and mediaId, returns a two field boolean object that says whether an object is in a queue or watched list
 - Throws 400 if any fields are missing

## Angular frontend
 - Refactors details pages to use new MediaItemInServlet
 - Adds pagination to queue component.

### CODE REVIEW

**Comment by TheGuyWhoCodes** [comment]
@henryksloan let me rename that variable before merging. it is in fact a page number, it was originally offset when implementing.

**Comment by henryksloan** [comment]
@TheGuyWhoCodes Alright, that makes sense.

File: src/main/java/com/google/sps/servlets/queue/MediaItemInServlet.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        response.getWriter().println(gson.toJsonTree(listResponse));
+    }
+
+    private boolean isInQueue(String userId, String mediaId) {
+        return !ofy().load().type(QueueListItemObject.class)
+                .filter("userId", userId)
+                .filter("mediaId", mediaId)
+                .order("-timestamp")
+                .list().isEmpty();
```
Comment (kirantumkur) [inline src/main/java/com/google/sps/servlets/queue/MediaItemInServlet.java:53]: Consider fetching just one item instead of list()

Comment (TheGuyWhoCodes) [reply src/main/java/com/google/sps/servlets/queue/MediaItemInServlet.java]: Fixed this with 1a83380. There is a .first() function that works great for this purpose!

**Pull request merged**